openweathermap-ios/openweathermap-ios/Maps/ViewControllers/MapViewController.swift

```diff
@@ -57,7 +62,8 @@ extension MapViewController: MKMapViewDelegate {
     }
     
     func mapView(_ mapView: MKMapView, regionDidChangeAnimated animated: Bool) {
-        updateAnnotation(location: mapView.centerCoordinate)
+        weatherManager.fetchWeather(latitude: mapView.centerCoordinate.latitude, longitude: mapView.centerCoordinate.longitude)
```

Comment:
I think you should add some validation here to avoid dispatch too many requests together.

Reply:
My control suggestion is to do so. But the problem is that I don't know where to change to "true" so that you can make a new request when changing the location.

```
if isFetchWeather {
            weatherManager.fetchWeather(latitude: mapView.centerCoordinate.latitude, longitude: mapView.centerCoordinate.longitude)
            isFetchWeather = false
        }
```

Reply:
Save the lastCenterCorordinate dispatched, and when the regionDidChange happens, check the distance between those two points. If the distance is greater than X meters, you dispatch again.